src/main/drivers/sdcard_spi.c

```diff
@@ -225,27 +227,30 @@ static uint8_t sdcard_sendCommand(uint8_t commandCode, uint32_t commandArgument)
         commands that require a CRC */
     };
 
-    uint8_t idleByte;
+    uint8_t idleByte1;
+    uint8_t idleByte2;
 
     // Note that this does not release the CS at the end of the transaction
     busSegment_t segments[] = {
+            {.u.buffers = {NULL, &idleByte1}, sizeof(idleByte1), false, sdcard_callbackIdle},
             {.u.buffers = {command, NULL}, sizeof(command), false, NULL},
-            {.u.buffers = {NULL, &idleByte}, sizeof(idleByte), false, sdcard_callbackNotIdle},
+            {.u.buffers = {NULL, &idleByte2}, sizeof(idleByte2), false, sdcard_callbackNotIdle},
             {.u.link = {NULL, NULL}, 0, true, NULL},
 
     };
 
-    if (!sdcard_waitForIdle(SDCARD_MAXIMUM_BYTE_DELAY_FOR_CMD_REPLY) && commandCode != SDCARD_COMMAND_GO_IDLE_STATE)
-        return 0xFF;
-
     sdcard.idleCount = SDCARD_MAXIMUM_BYTE_DELAY_FOR_CMD_REPLY;
 
     spiSequence(&sdcard.dev, &segments[0]);
 
     // Block pending completion of SPI access
     spiWait(&sdcard.dev);
 
-    return idleByte;
+    if ((idleByte1 != SDCARD_IDLE_TOKEN) && commandCode != SDCARD_COMMAND_GO_IDLE_STATE) {
+        return 0xFF;
```

Comment:
```suggestion
        return SDCARD_IDLE_TOKEN;
```

Reply:
0xFF is a failure code, and nothing to do with the idle status tokens.